downgrade

```diff
@@ -128,10 +132,48 @@ output_package() {
     done
   fi
 
-  printf "%s%3.4s) %s\\n" "$indicator" "$1" "$(
-    sed 's|https://.*/\([^/]\+\)$|\1 ('"$(gettext 'remote')"')|;
-         s|^/.*/\([^/]\+\)$|\1 ('"$(gettext 'local')"')|;' <<<"$2"
+  # Remote or local file
+  if [[ $path =~ ^/ ]]; then
+    location="$(gettext 'local')"
+  else
+    location="$(gettext 'remote')"
+  fi
+
+  # Remove the path, package name and extensions
+  # [epoch:]version-release-arch
```

Comment:
This bit is the only thing that concerns me. I think it could be done more simply, and I'm not sure it handles all possible kinds of `version` strings correctly. Before addressing that, we would need to get it wrapped in a unit test.

Can you please extract this logic to a dedicated function that accepts the path values on `stdin` and outputs separate words for the parts it's "parsing" out:

```sh
# Pick a better name please ;)
break_up_path()
  # your implementation here
}

# how you would use it:
read -r epoch version release arch < <(break_up_path <<< "$path")
```

Then wrap it in a unit test for all kinds of nasty `path` values?

Reply:
I skimmed through some repos and (apart from a small bug that is fixed now) it seems to work just fine.

Reply:
Gotcha. I'm pretty sure that if a package contained a `-` in its version string, this would break. I'm not sure if that's legal or not, but I can't find an existing example, so I'm good to not worry about it for now.